From b1bd391414b8c6cabc46a474dbba1b4119c5c593 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Tue, 29 Sep 2020 13:30:57 +0530
Subject: [PATCH 0911/1921] cn10k-cpt: Update CPT_RES_S as per new layout

Updates CPT_RES_S data structure as per new cn10k layout.

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I8c3591d24e2e2dfb5de8b878e29e8b1213a1df52
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/36932
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../crypto/marvell/cn10k/cn10k_cpt_hw_types.h    | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h b/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
index 23dd2a62ffac..af553526fe34 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
+++ b/drivers/crypto/marvell/cn10k/cn10k_cpt_hw_types.h
@@ -284,17 +284,19 @@ union cn10k_cpt_res_s {
 
 	struct {
 #if defined(__BIG_ENDIAN_BITFIELD) /* Word 0 - Big Endian */
-		u64 reserved_17_63:47;
-		u64 doneint:1;
+		u64 spi:32;
+		u64 rlen:16;
 		u64 uc_compcode:8;
-		u64 compcode:8;
+		u64 doneint:1;
+		u64 compcode:7;
 #else /* Word 0 - Little Endian */
-		u64 compcode:8;
-		u64 uc_compcode:8;
+		u64 compcode:7;
 		u64 doneint:1;
-		u64 reserved_17_63:47;
+		u64 uc_compcode:8;
+		u64 rlen:16;
+		u64 spi:32;
 #endif /* Word 0 - End */
-		u64 reserved_64_127;
+		u64 esn;
 	} s;
 };
 
-- 
2.31.1

